www/js/common/trip-detail.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+angular.module('emission.main.common.trip-detail',['ui-leaflet',
+                                      'ionic-datepicker',
+                                      'emission.services',
+                                      'emission.main.diary.services', 
+                                      'emission.main.common.services',
+                                      'nvd3'])
+
+.controller("CommonTripDetailCtrl", function($scope, $stateParams,
+                                        CommonGraph, $window) {
+    $scope.tripId = $stateParams.tripId;
+    $scope.trip = CommonGraph.data.cTripId2ObjMap[$scope.tripId];
+    var distributionStartHour = $scope.trip.common_hour_distribution;
+    var valsStartHour = [];
+    for (var i = 0; i < 24; i++) {
+      if (distributionStartHour.get(i)) {
+        valsStartHour.push({'label': i + ':00', 'value': distributionStartHour.get(i) });
+      } else {
+        valsStartHour.push({'label': i + ':00', 'value': 0 });
```

Comment:
Ah this is why. you need to get max and min appropriately and then stop pushing here.